<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChaserEditor</class>
 <widget class="QWidget" name="ChaserEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>622</width>
    <height>553</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chaser editor</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="10" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Run Order</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="m_loop">
        <property name="toolTip">
         <string>Run through over and over again</string>
        </property>
        <property name="text">
         <string>Loop</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_singleShot">
        <property name="toolTip">
         <string>Run through once and stop</string>
        </property>
        <property name="text">
         <string>Single Shot</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_pingPong">
        <property name="toolTip">
         <string>Switch direction at both ends</string>
        </property>
        <property name="text">
         <string>Ping Pong</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="6">
    <widget class="QToolButton" name="m_raise">
     <property name="toolTip">
      <string>Raise the selected step once</string>
     </property>
     <property name="text">
      <string notr="true">...</string>
     </property>
     <property name="icon">
      <iconset resource="qlcui.qrc">
       <normaloff>:/up.png</normaloff>:/up.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="5" column="6">
    <widget class="QToolButton" name="m_lower">
     <property name="toolTip">
      <string>Lower the selected step once</string>
     </property>
     <property name="text">
      <string notr="true">...</string>
     </property>
     <property name="icon">
      <iconset resource="qlcui.qrc">
       <normaloff>:/down.png</normaloff>:/down.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="3" column="6">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="11" column="0" colspan="6">
    <widget class="Line" name="line_3">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="2" column="6">
    <widget class="QToolButton" name="m_remove">
     <property name="toolTip">
      <string>Remove the selected step</string>
     </property>
     <property name="text">
      <string notr="true">...</string>
     </property>
     <property name="icon">
      <iconset resource="qlcui.qrc">
       <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="1" column="0" rowspan="9" colspan="6">
    <widget class="QTreeWidget" name="m_tree">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
     <property name="rootIsDecorated">
      <bool>false</bool>
     </property>
     <property name="itemsExpandable">
      <bool>false</bool>
     </property>
     <property name="allColumnsShowFocus">
      <bool>true</bool>
     </property>
     <property name="expandsOnDoubleClick">
      <bool>false</bool>
     </property>
     <column>
      <property name="text">
       <string>Step</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Function</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Fade In</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Fade Out</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Duration</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="1" column="6">
    <widget class="QToolButton" name="m_add">
     <property name="toolTip">
      <string>Add step(s) to the current position</string>
     </property>
     <property name="text">
      <string notr="true">...</string>
     </property>
     <property name="icon">
      <iconset resource="qlcui.qrc">
       <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="6" column="6" rowspan="4">
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0" colspan="7">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="m_nameLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Chaser name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="m_nameEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Name of the chaser being edited</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_copyButton">
       <property name="text">
        <string notr="true">...</string>
       </property>
       <property name="icon">
        <iconset resource="qlcui.qrc">
         <normaloff>:/editcopy.png</normaloff>:/editcopy.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_cutButton">
       <property name="text">
        <string notr="true">...</string>
       </property>
       <property name="icon">
        <iconset resource="qlcui.qrc">
         <normaloff>:/editcut.png</normaloff>:/editcut.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_pasteButton">
       <property name="text">
        <string notr="true">...</string>
       </property>
       <property name="icon">
        <iconset resource="qlcui.qrc">
         <normaloff>:/editpaste.png</normaloff>:/editpaste.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="10" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Direction</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QRadioButton" name="m_forward">
        <property name="toolTip">
         <string>Start from the first step</string>
        </property>
        <property name="text">
         <string>Forward</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="m_backward">
        <property name="toolTip">
         <string>Start from the last step</string>
        </property>
        <property name="text">
         <string>Backward</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Maximum</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="10" column="2">
    <widget class="QGroupBox" name="m_fadeInGroup">
     <property name="title">
      <string>Fade In Speed</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <widget class="QRadioButton" name="m_fadeInCommonRadio">
        <property name="toolTip">
         <string>All steps have common fade in speed set by the chaser</string>
        </property>
        <property name="text">
         <string>Common</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="m_fadeInPerStepRadio">
        <property name="toolTip">
         <string>Each step has its own fade in speed set by the chaser</string>
        </property>
        <property name="text">
         <string>Per Step</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="m_fadeInDefaultRadio">
        <property name="toolTip">
         <string>Use each function's own fade in speed</string>
        </property>
        <property name="text">
         <string>Default</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="10" column="3">
    <widget class="QGroupBox" name="m_fadeOutGroup">
     <property name="title">
      <string>Fade Out Speed</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="0">
       <widget class="QRadioButton" name="m_fadeOutCommonRadio">
        <property name="toolTip">
         <string>All steps have common fade out speed set by the chaser</string>
        </property>
        <property name="text">
         <string>Common</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="m_fadeOutPerStepRadio">
        <property name="toolTip">
         <string>Each step has its own fade out speed set by the chaser</string>
        </property>
        <property name="text">
         <string>Per Step</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="m_fadeOutDefaultRadio">
        <property name="toolTip">
         <string>Use each function's own fade out speed</string>
        </property>
        <property name="text">
         <string>Default</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="10" column="4">
    <widget class="QGroupBox" name="m_durationGroup">
     <property name="title">
      <string>Step Duration</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="1" column="0">
       <widget class="QRadioButton" name="m_durationCommonRadio">
        <property name="toolTip">
         <string>All steps have common step duration set by the chaser</string>
        </property>
        <property name="text">
         <string>Common</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="m_durationPerStepRadio">
        <property name="toolTip">
         <string>Each step has its own duration set by the chaser</string>
        </property>
        <property name="text">
         <string>Per Step</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Maximum</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="12" column="0" colspan="3">
    <widget class="QPushButton" name="m_testButton">
     <property name="toolTip">
      <string>See what the chaser does when it is run</string>
     </property>
     <property name="text">
      <string>Test Run</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="12" column="3">
    <widget class="QPushButton" name="m_testPreviousButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>Skip to the previous step</string>
     </property>
     <property name="text">
      <string>Previous</string>
     </property>
    </widget>
   </item>
   <item row="12" column="4">
    <widget class="QPushButton" name="m_testNextButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>Skip to the next step</string>
     </property>
     <property name="text">
      <string>Next</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="qlcui.qrc"/>
  <include location="qlcui.qrc"/>
 </resources>
 <connections/>
</ui>
